### ISSUE TITLE
Does not compile

### ISSUE THREAD

**Issue by ligi** [mgrzechocinski/dagger2-example#3]
➜  dagger2-example git:(master) gradle clean build
Parallel execution is an incubating feature.
:app:clean
:app:preBuild UP-TO-DATE
:app:preDebugBuild UP-TO-DATE
:app:compileDebugNdk UP-TO-DATE
:app:checkDebugManifest
:app:preReleaseBuild UP-TO-DATE
:app:prepareComAndroidSupportAppcompatV72200Library
:app:prepareComAndroidSupportSupportV42200Library
:app:prepareDebugDependencies
:app:compileDebugAidl
:app:compileDebugRenderscript
:app:generateDebugBuildConfig
:app:generateDebugAssets UP-TO-DATE
:app:mergeDebugAssets
:app:generateDebugResValues UP-TO-DATE
:app:generateDebugResources
:app:mergeDebugResources
/home/ligi/tmp/dagger2-example/app/build/intermediates/exploded-aar/com.android.support/appcompat-v7/22.0.0/res/drawable-hdpi-v4/abc_spinner_mtrl_am_alpha.9.png: libpng warning: iCCP: Not recognizing known sRGB profile that has been edited

:app:processDebugManifest
:app:processDebugResources
:app:generateDebugSources
:app:compileDebugJava
:app:patchAndroidJar
:app:_compileDebugJava
/home/ligi/tmp/dagger2-example/app/src/main/java/net/grzechocinski/android/dagger2example/D2EBaseApplication.java:8: error: cannot find symbol
import net.grzechocinski.android.dagger2example.internal.di.Dagger_D2EComponent;
                                                           ^
  symbol:   class Dagger_D2EComponent
  location: package net.grzechocinski.android.dagger2example.internal.di
/home/ligi/tmp/dagger2-example/app/src/debug/java/net/grzechocinski/android/dagger2example/internal/di/DebugDependenciesModule.java:10: error: @Provides methods can only be present within a @Module
    DebugDependency provideDebugDependency() {
                    ^
/home/ligi/tmp/dagger2-example/app/src/debug/java/net/grzechocinski/android/dagger2example/internal/di/D2EComponent.java:12: error: net.grzechocinski.android.dagger2example.internal.di.DebugDependenciesModule is listed as a module, but is not annotated with @Module
public interface D2EComponent extends D2EGraph{
       ^
3 errors
:app:_compileDebugJava FAILED

FAILURE: Build failed with an exception.
- What went wrong:
  Execution failed for task ':app:_compileDebugJava'.
  
  > Compilation failed; see the compiler error output for details.
- Try:
  Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 23.5 secs

**Comment by mgrzechocinski** [comment]
Fixed by #5

**Comment by chornge** [comment]
I was dealing with the same issue, the error was coming from referencing the wrong class:
```
@Component(modules = CustomClassComponent.class)
public interface ...
```

The annotated interface above references itself, instead of a module class. Check to see that your annotated component interface has a module reference, not a component reference:

```
@Component(modules = CustomClassModule.class)
public interface ...
```
